Tests/NIOTests/SocketChannelTest.swift

```diff
@@ -577,9 +577,9 @@ public final class SocketChannelTest : XCTestCase {
             // Build server channel; after this point the server called listen()
             let serverPromise = group.next().makePromise(of: IOError.self)
             let serverChannel = try assertNoThrowWithValue(ServerBootstrap(group: group)
-                .serverChannelOption(ChannelOptions.socketOption(.reuseaddr), value: 1)
-                .serverChannelOption(ChannelOptions.backlog, value: 256)
-                .serverChannelOption(ChannelOptions.autoRead, value: false)
+                .serverChannelOptions([.allowImmediateEndpointAddressReuse,
```

Comment:
just to explicitly pointing this out to @Lukasa & @glbrntt : If merged as is, we will have to ways of spelling channel options:

- `serverChannelOption`: the old way
- `serverChannelOptions` (mind the all important `s`): the convenience option way

I think this is confusing. I would've expected that if I go from singular to plural, I would be able to use the same kind of options. But that's not the case in the proposed API.

Reply:
Yeah, may be worth bike shedding the name.

Reply:
I think it's _slightly_ confusing. A better name would be great but I haven't be able to come up with any good suggestions; adding things like "shorthand" or "convenience" to the name feels clunky imho.

Reply:
We want this to feel like the most natural name if it's the one we want to encourage so adding extra words feels wrong.
I see where you're coming from but I don't think we're likely to come up with anything better.

Reply:
`serverOptions` / `clientOptions`?

Reply:
and just `options` for the client channels?

Reply:
WFM.